stl/inc/tuple

```diff
@@ -419,9 +419,18 @@ public:
         : tuple(_Unpack_tuple_t{}, _STD move(_Right)) {}
 #endif // ^^^ !_HAS_CONDITIONAL_EXPLICIT ^^^
 
+#if _HAS_CONDITIONAL_EXPLICIT
+    template <class _Alloc, class _This2 = _This,
+        enable_if_t<conjunction_v<is_default_constructible<_This2>, is_default_constructible<_Rest>...>, int> = 0>
+    explicit(
+        !conjunction_v<_Is_implicitly_default_constructible<_This2>, _Is_implicitly_default_constructible<_Rest>...>)
+        tuple(allocator_arg_t, const _Alloc& _Al)
+        : _Mybase(allocator_arg, _Al), _Myfirst(_Al, allocator_arg) {}
+#else // ^^^ _HAS_CONDITIONAL_EXPLICIT ^^^ / vvv !_HAS_CONDITIONAL_EXPLICIT vvv
     template <class _Alloc, class _This2 = _This,
         enable_if_t<conjunction_v<is_default_constructible<_This2>, is_default_constructible<_Rest>...>, int> = 0>
     tuple(allocator_arg_t, const _Alloc& _Al) : _Mybase(allocator_arg, _Al), _Myfirst(_Al, allocator_arg) {}

```

Comment:
Thanks! The `_HAS_CONDITIONAL_EXPLICIT` codepath looks completely correct to me. However, you should also update the `!_HAS_CONDITIONAL_EXPLICIT` codepath, following the same pattern as the other constructors. (That is, you need to split this into two overloads, one implicit and one `explicit`, with appropriately adjusted constraints.)

You can test this by compiling with `/D_HAS_CONDITIONAL_EXPLICIT=0`. (Otherwise, only NVIDIA's CUDA compiler activates it.)